clippy_lints/src/methods/mod.rs

```diff
@@ -107,6 +108,42 @@ declare_clippy_lint! {
     "used `cloned` where `copied` could be used instead"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `_.cloned().<func>()` where call to `.cloned()` can be postponed,
+    /// or often removed entirely.
```

Comment:
Don't we issue `REDUNDANT_CLONE` when the `.cloned()` can be removed?

Reply:
@llogiq Ok, fixed.